kyo-prelude/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -259,6 +259,46 @@ sealed abstract class Stream[V, -S]:
         ))
     end changes
 
+    /** Transforms the stream by regrouping elements into chunks of the specified size.
+      *
+      * This operation maintains the order of elements while potentially redistributing them into new chunks. Smaller chunks may occur in
+      * two cases:
+      *   - When there aren't enough remaining elements to form a complete chunk
+      *   - When the input stream emits an empty chunk
+      *
+      * @param chunkSize
+      *   The target size for each chunk. Must be positive - negative values will be treated as 1.
+      * @return
+      *   A new stream with elements regrouped into chunks of the specified size
+      */
+    def rechunk(chunkSize: Int)(using tag: Tag[Emit[Chunk[V]]], frame: Frame): Stream[V, S] =
+        Stream.sized[V, S](chunkSize)(cs =>
+            ArrowEffect.handleState(tag, Chunk.empty[V], emit.andThen(Emit(Chunk.empty[V])))(
+                [C] =>
+                    (input, buffer, cont) =>
+                        if input.isEmpty && buffer.nonEmpty then
+                            Emit.andMap(buffer)(ack => (Chunk.empty, cont(ack)))
+                        else
+                            val combined = Chunk.from(buffer.appendedAll(input))
```

Comment:
It seems this could be
```suggestion
                            val combined = buffer.concat(input)
```